### ISSUE TITLE
hello again

### ISSUE THREAD

**Issue by themirrazz** [kelbazz/Kelbazz#2]
Hey, do you still have that profile picture you made me?

**Comment by kelbazz** [comment]
Oh yeah but how can i send it to you ?

**Comment by themirrazz** [comment]
Just send it through here :)

**Comment by themirrazz** [comment]
Also I have a new website now https://themirrazz.repl.co/

**Comment by kelbazz** [comment]
Nice. And yeah i'm dumb

**Comment by kelbazz** [comment]
I'll send it here

**Comment by themirrazz** [comment]
hi

**Comment by themirrazz** [comment]
btw

**Comment by themirrazz** [comment]
(awkward silence, github edition)

**Comment by kelbazz** [comment]
🥴 hi

**Comment by kelbazz** [comment]
![TM logo](https://user-images.githubusercontent.com/81097895/224405861-6261d1af-eb84-424b-a8d1-51c1f7aed809.png)

**Comment by kelbazz** [comment]
tada !

**Comment by kelbazz** [comment]
Do you want the svg version too ?

**Comment by themirrazz** [comment]
sure, thanks 🥴
Gosh even Google Woozy looks okay-ish - is Apple the only one with a terrible woozy? Also https://themirrazz.github.io/woozyness/pumpkin.html

**Comment by themirrazz** [comment]
you'll need https://rhino3d.com/ - get the evaluation license

**Comment by themirrazz** [comment]
its free :)

**Comment by themirrazz** [comment]
perfect :) thanks again
![image](https://user-images.githubusercontent.com/89490659/224424602-9cea48fe-ef84-413a-92d9-40d89bfee560.png)

**Comment by themirrazz** [comment]
also is this mobile UI good
![image](https://user-images.githubusercontent.com/89490659/224424700-bb38fb4a-e966-4b71-ab82-fe16f4f1de86.png)
![image](https://user-images.githubusercontent.com/89490659/224424713-6d401e5d-ff97-45bc-b512-81658ea8dd45.png)

**Comment by kelbazz** [comment]
Honestly... Err... well looks a bit outdated or old but that's maybe intentional

**Comment by kelbazz** [comment]
Here is the svg btw

**Comment by kelbazz** [comment]
![TM logo](https://user-images.githubusercontent.com/81097895/224429648-d71d2b01-3024-40a3-8906-4e2545d48dca.svg)

**Comment by kelbazz** [comment]
Oh and also didn't know if i told you but i'm redesigning Border from "scratch"

**Comment by kelbazz** [comment]
![image](https://user-images.githubusercontent.com/81097895/224430077-a4d19921-977e-470c-8f8d-20df94f2871d.png)
This will be the final result

**Comment by kelbazz** [comment]
And this what i've done yet:
![image](https://user-images.githubusercontent.com/81097895/224430243-f6428568-2ca5-4e9b-a2b2-5555b36ef9d5.png)

**Comment by kelbazz** [comment]
Also for border desktop. After a long reflection I think creating it with electron is not a bad idea as there's no easy alternative and for the MV3 we could simply create our own ad blocker and as we'll have our proper extension system "brx" the plugin system will not be a problem anymore. Of course Border is now also your project so i won't create anything without your consent.

**Comment by kelbazz** [comment]
Also here's what mobile UI can be with the new design:
![image](https://user-images.githubusercontent.com/81097895/224433786-d6c9042a-3345-4d9e-b5af-fe7a1719fa28.png)

**Comment by themirrazz** [comment]
Woah no fricking way! Suggestion: make it so people can move around the buttons, like in Firefox! I could help build that :)

**Comment by themirrazz** [comment]
it looks too nice!!

**Comment by themirrazz** [comment]
Also looks like `border.org` domain is available on GoDaddy and `borderbrowser.org` is just... available!

**Comment by kelbazz** [comment]
Well if you can teach me how to move thing hehe why not

**Comment by themirrazz** [comment]
Also I'm starting to have mixed feelings on the refresh button being in between the back/forward.
tl;dr: I like it but still personally prefer the back and forward buttons being next to each other on desktop, probably because every desktop browser I use has the refresh button after back/forward buttons

**Comment by themirrazz** [comment]
Also the tab title on mobile in general makes it a bit big when shown alongside the page, just so you know. As in it might not fit ideally on all screens, especially on phones in landscape mode.

**Comment by themirrazz** [comment]
if you want

**Comment by kelbazz** [comment]
Yeah why not, if you can... But nice! I'll modify the concept for the mobile version and we're good. I'm just stuck on how to make the iframe part. As I wan't it to be customizable but also a native part of the browser :/

**Comment by themirrazz** [comment]
oh I have an idea!

**Comment by themirrazz** [comment]
Take a look at https://bordermobile.awesomecrater.repl.co/

**Comment by themirrazz** [comment]
Create a new tab, visit border://colors, change the scheme, switch to first New Tab... it has the old colors because it isn't transparent! (Though v1.11 features Dynamic UI Updating)

**Comment by themirrazz** [comment]
I can't wait for v1.11

**Comment by themirrazz** [comment]
Oh you know how the icons are SVG instead of font now? I'm thinking of using a new BRX-based theme format (with a "manifest.json" file) where themes can have custom icons.
Border: * fails the "try not to be the most customizable browser" challenge *

**Comment by themirrazz** [comment]
lol

**Comment by themirrazz** [comment]
(joke)
what if it looked like this
![image](https://user-images.githubusercontent.com/89490659/227066148-b0fd578b-c07c-4fce-b8f8-700ac779791d.png)

**Comment by kelbazz** [comment]
XD, colors are pretty cools, for themes i still have to implement the base theme system lol but i'll make it a bit like the w96 one (with a manifest)

**Comment by themirrazz** [comment]
lol can you send me the exec

**Comment by kelbazz** [comment]
Well idk if i can here but the code is on codeberg: https://codeberg.org/Border/w96

**Comment by kelbazz** [comment]
Border 1.11 will be much more structured a the bin won't repair itself for now. There will probably be a bin for that. But yeah i tried a different workflow. Also let me add you to this repo

**Comment by kelbazz** [comment]
Also ! I'm working on a Windows 96 Runtime That uses Web Workers as context so you can easely terminate them, control their access to the api (for permissons) and use ES modules.

**Comment by themirrazz** [comment]
Noice

**Comment by kelbazz** [comment]
Also: https://git.sys36.net/windows-96/sys-userland/

**Comment by kelbazz** [comment]
Here's the new "workflow":
![image](https://user-images.githubusercontent.com/81097895/232308442-eeb07194-1d56-49bb-b035-51caac9bd2a9.png)

**Comment by themirrazz** [comment]
woah thats complicated lol

**Comment by kelbazz** [comment]
Yeah, "*organised*" lmao. Also as the structure is now stored in html files, the UI is now customisable so you can add buttons etc... Plus, we now have "system blugins" they act like blugins but are ran before other blugins and it is possible to run Border in safe mode to disable user blugins at startup: `border --safe`.

**Comment by kelbazz** [comment]
The history system is ready and i don't use `window.history` anymore to avoid conflict with the browser window. Aaaand `config.json` is now `config.yaml` so if advanced users wanna play with it (even if we'll have a UI for options) they can do it with a simpler syntax than json lol. Now i just have to rebuild the bookmark system. I'm so hyped hehe.

**Comment by themirrazz** [comment]
Nice, also should we make it so you can switch whether the refresh button or go forward button comes first on desktop as most people are use to "back, forward, reload" and some might prefer that, or we could make it so you can drag the UI elements to more places than firefox lets you, but the bookmarks bar inside of each tab and drag the close/max/min to the bottom lol maybe

**Comment by kelbazz** [comment]
¯\_(ツ)_/¯ i can try the option to switch the buttons but for all the UI i think i'll let blugins and themes do the job

**Comment by themirrazz** [comment]
Yeah, but themes its hard to since its moving around, and position can be a bit buggie

**Comment by kelbazz** [comment]
> Yeah, but themes its hard to since its moving around, and position can be a bit buggie

Well with the new version you can actually change the html of border as it is a web page. So i guess you can just change the two buttons and it will work.

**Comment by themirrazz** [comment]
I built Firefox from source, lots of it is literally just HTML/CSS/JS and `.ftl` files for localization, we could maybe have Firefox-based Border

**Comment by themirrazz** [comment]
Hi, again!
I drew some concept of Border on iOS and Android ![image](https://github.com/kelbazz/Kelbazz/assets/89490659/46188848-f794-423d-9fb0-c6959b30ef6e)
![image](https://github.com/kelbazz/Kelbazz/assets/89490659/590df34e-592f-452f-aa77-0b3bc5f1bc2c)
Just need to finish the desktop concepts

**Comment by themirrazz** [comment]
> Oh why not ! I just hope i can build it too without needing too many permissions cause i'm not admin

It requires admin permissions iirc as it needs to make C:\mozilla-build and C:\mozilla-source, but actually I think it doesn’t need admin to make those.

**Comment by kelbazz** [comment]
Wow sorry to reply this late :/ I've not checked this issue for a while now. The concept is great but i think the three dot menu should be the middle one

**Comment by themirrazz** [comment]
okay

**Comment by themirrazz** [comment]
also do you think you could make a 128x128 icon for a Discord server I'm making called themirrazz City

**Comment by themirrazz** [comment]
actually hold off on that

**Comment by kelbazz** [comment]
So we can talk on discord?

**Comment by themirrazz** [comment]
sure

**Comment by themirrazz** [comment]
just send me a friend request

**Comment by kelbazz** [comment]
Can i have your discord tag ?

**Comment by kelbazz** [comment]
ok nvm it was easy lol

**Comment by kelbazz** [comment]
![image](https://github.com/kelbazz/.profile/assets/81097895/d1df2b47-a399-45df-ac2e-e2aee8930a51)

**Comment by themirrazz** [comment]
I was talking about the one based off of the pictures I sent you, lol
Wasn't sure whether you had made them or not

**Comment by kelbazz** [comment]
Err... Wait which one

**Comment by kelbazz** [comment]
Let me check

**Comment by themirrazz** [comment]
k

**Comment by themirrazz** [comment]
Oh also at https://new.themirrazz.repl.co/projects/
![image](https://github.com/kelbazz/.profile/assets/89490659/f1e9409c-e670-4c40-aac1-1c5ca9ab3976)

**Comment by kelbazz** [comment]
Well it's the v2.0 as everything is remade from the ground up

**Comment by kelbazz** [comment]
And i'm really sorry but i cannot work on your logo rn as my pc is locked :/ I'll do it as soon as i can use it

**Comment by kelbazz** [comment]
[TM_logos.zip](https://github.com/kelbazz/.profile/files/12537560/TM_logos.zip)

Hey mirrazz ! As promised, your logos are here.
Hope you'll like them!

**Comment by themirrazz** [comment]
thanks!

**Comment by themirrazz** [comment]
btw do you have the SVGs for them?

**Comment by kelbazz** [comment]
Yeah for sure! Just let me grab them

**Comment by themirrazz** [comment]
ok thanks

**Comment by kelbazz** [comment]
[Miscs (2).zip](https://github.com/kelbazz/.profile/files/12553407/Miscs.2.zip)

**Comment by themirrazz** [comment]
thanks

**Comment by themirrazz** [comment]
7 words, one URL, and 97 characters: Border Labs. https://github.com/Border-Labs/readme/tree/main

**Comment by themirrazz** [comment]
the Labs release channel of Border will have both versions of v1.10 and v1.11 - v1.10 will be recommended for labs since even the stable release of v1.11 for desktop will probably be a little unstable if we make it with Firefox.
We could possibly make it Electron, but some things might be a bit unstable.

One thing I want to do in the Labs release of v1.10 is implement this stupid API I made: The Web Chaos API.
Detect Chaos Energy from your website. Supported as of sep 13 2023 in 0% of browsers. Very non-standard. Requires specialized hardware that people don't yet know how to make (except maybe SEGA).
(Demo at https://new.themirrazz.repl.co/ChaosDetect.html)

The second is the experimental Border Sync server (which will be called Border Labs Sync). It will allow you to see in real-time what tabs are open on your other devices as well as sync bookmarks, color schemes, and certain settings.
I'm not sure if seeing what tabs are open will be included in the regular version of Border Sync.

Do you have any thoughts for features of the Border Labs release channel?

**Comment by kelbazz** [comment]
What exactly does this Web Chaos API? But except that, i really like the fact that we have a separated channel just for the lab/canary version! I've some plans to make a kind of system to be able to add bindings to Border so it can compile to w96 and electron more easily. Plus i'm refactoring everything to be compatible with the wdt so we can use it to compile the code for both platform so it's easier and faster as the bins will be smaller (I remember the time where Border was one Big file lol). So yeah pretty cool. Plus it's a really cool idea to have some special APIs just for Border. The only problem for now is how to be able to have a good way to communicate between both the site and Border (especially with cors iframes). I hope you will be able to help me for future development.

**Comment by themirrazz** [comment]
1. The Web Chaos API detects Chaos Energy. It's the world's stupidest API, as it's not like a Chaos Emerald is going to fall out of the sky.
2. Can I credit you for making my icon?

**Comment by kelbazz** [comment]
1. Lmao well idk for this API but yeah we could add a Easter API with some fun features like confettis and other stuff.
2. And yes for sure!

**Comment by themirrazz** [comment]
okay lol

**Comment by themirrazz** [comment]
btw Check out my scratch
![image](https://github.com/kelbazz/.profile/assets/89490659/de6bfc35-0db4-4950-9532-d9d5a3973ca5)
![image](https://github.com/kelbazz/.profile/assets/89490659/005c7236-52ae-4f4b-9e4b-ec5ec6aafe46)

**Comment by themirrazz** [comment]
(real)

**Comment by themirrazz** [comment]
Hey, would it be okay if I add ads (from my self-hosted ad service) to the Border News page? I can pay you if you want, I just have to make a PayPal account, the ad service itself doesn't track clicks (though the ads themselves might track that the click came from an ad on KusoAds! using utm_xxxxx parameters, but that's implemented by the website owners themselves), it doesn't lag or take up all your RAM (tested - it doesn't use up a bunch of your CPU like Google AdSense or other popular online ads), the only ad is for my YouTUbe channel as of right now because I haven't convinced anybody to buy an ad slot yet, selecting which types of ads you'll choose is all stored in localStorage (not cookies) and does not sync to any account

**Comment by kelbazz** [comment]
Hi! Soo, this is quite a question. I don't really know for now. I'm not against having a revenue part from an open source project but if we include ads it has to be clean: Not too visible, user friendly, and not an obligation. We shouldn't block the site for ppl with an ad block in my opinion. But that's my point of view so yeah. Tough question 🤣. Also i think we should redesign the website for the v2. We're far from the end But it's progressing. I've fixed some bugs, rewritten the default css with css nesting and i'm now using WDT for compilation so this is faster and better. And this also means that you can build Border from the source code directly in Windows 96! That's mind blowing lmao!

**Comment by themirrazz** [comment]
> I'm not against having a revenue
Currently, there is no revenue as nobody has bought an ad slot yet (there is an ad for my YouTube channel though, since ads can be region restricted to certain regions and I haven't added proper error handling for if there is no other ads).
Also paying for placement on another site is kinda hard since I made my ad system use a pay-per-time basis (starting at $2 for 1 week going up to $103 for 1 year) rather than a pay-per-click basis.

> Not too visible
Do banner ads work? Because that's what my system uses and it doesn't support any othe ads. There could be an option to turn them off in the menu. I'm just too lazy (for right now) to make my ad system support anything besides banner ads.
I also don't thinks that ads should be on the Border homepage, or on the themes/blugins download pages, except maybe for things like "sponsored themes" or "sponsored blugins" (which would be a completely separate system).

> User friendly
Yeah, I have guidelines for the advertisements to make sure they're not too inappropriate, and I'm setting up the region lock based on geolocation using Geo.JS (currently, country region lock only has United States, United Kingdom, Japan, China, and Germany, and currently region is set using URL parameters - the GeoJS system is relatively new). The AdChoices button should look familiar unless I drew the logo wrong, and then the "X" button hides the ad to go to the next one (if continue showing more ads after hidden is enabled), and the "!" button reports inappropriate ads.

> We shouldn't block the site for ppl with an ad block in my opinion
Yeah I wasn't planning on it, don't do that on my website's "news" section either. While currently, my ad system isn't blocked by adblockers as of right now, it's relatively easy to add them by just adding it by domain (kusoadhost.themirrazz.repl.co), and the Border website could, as stated above, have a switch to disable them.
I personally think that a dialog saying like "Hey, please consider disabling your ad blocker to support us" is fine, but blocking access completely to people who have one isn't fine.

> Also i think we should redesign the website for the v2.
Great idea!

> and i'm now using WDT for compilation so this is faster and better. And this also means that you can build Border from the source code directly in Windows 96!
wow

> That's mind blowing lmao!
*/j* The surgeons have finally put my brain back together (joke - it never actually got blown apart in the first place)

**Comment by kelbazz** [comment]
Ok i think we're good! Do you just have a website where i can test the ad itself ? Like a demo or something.

**Comment by kelbazz** [comment]
![image](https://github.com/kelbazz/.profile/assets/81097895/bd17c9c6-ee54-44ee-8914-338b65ad20d6)
Progressing...

**Comment by themirrazz** [comment]
> Ok i think we're good! Do you just have a website where i can test the ad itself ? Like a demo or something.

Just use my News page https://new.themirrazz.repl.co/news/new-chromeos-117-crazy-error-in-the-works/

**Comment by themirrazz** [comment]
> ![image](https://user-images.githubusercontent.com/81097895/275423134-bd17c9c6-ee54-44ee-8914-338b65ad20d6.png) Progressing...

Nice

**Comment by themirrazz** [comment]
btw have you seen this yet https://www.youtube.com/watch?v=SjvssIpDauY

Warning: contains... mild? swearing and hurt apples (slight scar)

**Comment by kelbazz** [comment]
Now i did!

**Comment by themirrazz** [comment]
> Now i did!
lol thoughts

**Comment by kelbazz** [comment]
Firstly, your new website is really cool! The scroll animations are smooth and nice. I could try to optimize even though i'm not really used to working with those kinds of parallax scrolling effects. Where could I find the source code?

**Comment by themirrazz** [comment]
Uh, lol just view the source of the website, it's in the file /index-beta.js and /snowflake.js and /index-beta.css and /sunofureikku.css

**Comment by themirrazz** [comment]
Also in "/" on the ja-JP version but "/index-beta.html" on the en-US version
